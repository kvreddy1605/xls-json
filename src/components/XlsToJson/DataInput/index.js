import React from 'react';
import './index.scss';

class DataInput extends React.Component {
	constructor(props){
		super(props);
		this.handleChange = this.handleChange.bind(this);
	}

	handleChange(e){
		const files = e.target.files;
		if(files && files[0]) this.props.handleFile(files[0]);
	};

	render() { 
		return (
			<div className="dataInput-container" >
				<form>
					<label>Browse a file with type xls or csv.</label> <br />
					<input type="file" id="file" accept=".xls, .xlsx, .csv, .xlsb, .xlsm" onChange={this.handleChange} />
				</form>
			</div>
		); 
  	}
}

export default DataInput;