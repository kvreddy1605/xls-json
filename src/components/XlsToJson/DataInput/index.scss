.dataInput-container{
    input{
        margin: 20px;
    }
}