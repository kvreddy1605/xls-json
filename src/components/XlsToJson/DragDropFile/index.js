import React from 'react';
import './index.scss';

class DragDropFile extends React.Component {
	constructor(props){
		super(props);
		this.onDrop = this.onDrop.bind(this);
		this.suppress = this.suppress.bind(this);
	}

	onDrop(evt){ 
		evt.stopPropagation(); 
		evt.preventDefault();
		const files = evt.dataTransfer.files;
		if(files && files[0]) this.props.handleFile(files[0]);
	};

	suppress(evt){ 
		evt.stopPropagation(); 
		evt.preventDefault(); 
	};

	render() { 
		return (
			<div className="dragDropFile-container" onDrop={this.onDrop} onDragEnter={this.suppress} onDragOver={this.suppress}>
				{this.props.children}
			</div>
		);  
	}
};

export default DragDropFile;