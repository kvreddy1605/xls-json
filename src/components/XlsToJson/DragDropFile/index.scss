.dragDropFile-container{
    width: 400px;
    height: 200px;
    border: 2px solid #888;
    border-radius: 5px;
    padding: 10px;
}