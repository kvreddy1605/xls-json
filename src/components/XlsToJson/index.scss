.xlsToJson-container{
    .invalidMsg {
        color : red;
        font-size: 12px;
        font-weight: bold;
    }

    .watermark {
        color: #d0d0d0;
        font-size: 20px;
    }

}

