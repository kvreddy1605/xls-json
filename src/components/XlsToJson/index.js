import XLSX from 'xlsx';
import React from 'react';
import DragDropFile from './DragDropFile';
import DataInput from './DataInput';
import './index.scss';

class XlsToJson extends React.Component {
	constructor(props){
		super(props);
		this.state = {
			data: [],
			showInvalidMsg : false,
			fileName: ""
		}
		this.handleFile = this.handleFile.bind(this);
	}
	
	handleFile(file){
		//File Validation
		if( !this.validateFileExtension(file.name) ) { console.log("1"); return; }

		//Parsing File data into JSON
		const reader = new FileReader();
		const rABS = !!reader.readAsBinaryString;
		reader.onload = (e) => {
			const bstr = e.target.result;
			const wb = XLSX.read(bstr, {type:rABS ? 'binary' : 'array'});
			const wsname = wb.SheetNames[0];
			const ws = wb.Sheets[wsname];
			const data = XLSX.utils.sheet_to_json(ws, {header:1});  /* Array of Rows [ [A1, B1, C1], [A2, B2, C2]] */ 
			this.setState({ data: data }, () => this.props.onJSONOutput(this.state.data));
		};
		if(rABS) reader.readAsBinaryString(file); else reader.readAsArrayBuffer(file);			
	};

	validateFileExtension(fileName){
		if(/(\.xls|\.xlsx|\.csv|\.xlsb|\.xlsm)$/i.exec(fileName)){
			this.setState({ showInvalidMsg: false, fileName: fileName });
			return true;
		}
		else{
			this.setState({ showInvalidMsg: true, fileName: "" });
			return false;
		}
	}
	
	render() {
		let selectedFileName = <p className="watermark" >Drag and Drop a file here.</p>;
		if(this.state.showInvalidMsg){
			selectedFileName = <p className="invalidMsg" >Invalid File Type Selected! Please select a valid file type.</p>
		}
		if(this.state.fileName){
			selectedFileName = <p>Chosen File Name: {this.state.fileName}</p>
		}

		return (
			<div className="xlsToJson-container">
				<DragDropFile handleFile={this.handleFile}>
					<DataInput handleFile={this.handleFile} />
					{ selectedFileName }
				</DragDropFile>
			</div>
		); 
	}
};


export default XlsToJson;