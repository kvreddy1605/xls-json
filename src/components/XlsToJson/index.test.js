import React from 'react';
import XlsToJson from './index';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('XlsToJson Component Test', () => {

    it('SnapShot Test', () => {
        const wrapper = shallow(<XlsToJson onJSONOutput={() => null} />)
    
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('testing validateFileExtension method', () => {
        const wrapper = shallow(<XlsToJson onJSONOutput={() => null} />)
        
        expect(wrapper.instance().validateFileExtension('test.xls')).toBe(true);
        expect(wrapper.instance().validateFileExtension('test.xls.txt')).toBe(false);
    })
})




