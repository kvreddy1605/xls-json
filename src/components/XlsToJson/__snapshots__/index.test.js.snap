// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XlsToJson Component Test SnapShot Test 1`] = `
<div
  className="xlsToJson-container"
>
  <DragDropFile
    handleFile={[Function]}
  >
    <DataInput
      handleFile={[Function]}
    />
    <p
      className="watermark"
    >
      Drag and Drop a file here.
    </p>
  </DragDropFile>
</div>
`;
